src/forcing/ForcingsEngine.cpp

```diff
@@ -0,0 +1,239 @@
+#include "ForcingsEngine.hpp"
+
+namespace data_access {
+
+std::unordered_map<std::string, std::unique_ptr<ForcingsEngine>> ForcingsEngine::instances_{};
+
+ForcingsEngine::ForcingsEngine(const std::string& init, size_type time_start, size_type time_end)
+  : time_start_(std::chrono::seconds{time_start})
+  , time_end_(std::chrono::seconds{time_end})
+{
+    check_runtime_requirements();
```

Comment:
It looks like the things checked in there only need to happen once per run, not once per instance created